rpcs3/Emu/Cell/SPUCommonRecompiler.cpp

```diff
@@ -7363,16 +7386,34 @@ spu_program spu_recompiler_base::analyse(const be_t<u32>* ls, u32 entry_point, s
 			value.reg2 = pattern.reg2;
 		}
 
-		if (g_cfg.core.spu_accurate_reservations)
+		bool allow_pattern = true;
+
+		if (g_cfg.core.spu_accurate_reservations) if (0)
 		{
-			// Because enabling it is a hack, as it turns out
-			// continue;
+			static constexpr std::initializer_list<std::string_view> allowed_patterns =
+			{
+				"620oYSe8uQqq9eTkhWfMqoEXX0us"sv
```

Comment:
Where does this come from?

Reply:
It's the hash of the cellspurs jobchain pattern in #17733 
The problem with PUTLLC16 optimization, that it is in theory correct at the bounds of the spu function.
But if the SPU code reuses the cache line data observed, it is not truly atomic.
So we may enable it only for known cases where SPU atomic data is not used after the function leaves.

Tl;dr:

1. Atomic compare exchange 16 bytes operation. (rest of data is not read)  -> good for RPCS3 to optimize.
2. Fetch 128 bytes (read them later), modify only 16bytes. -> Bad for RPCS3 to optimize.

This difference cannot be known at analyzer time but from observing callers.